varnish-sys/src/vcl/str_or_bytes.rs

```diff
@@ -0,0 +1,34 @@
+use std::str::from_utf8;
+
+#[derive(Debug)]
+pub enum StrOrBytes<'a> {
+    Utf8(&'a str),
+    Bytes(&'a [u8]),
+}
```

Comment:
I'm wondering, should we have a `None` variant here? That would simplify a bunch of interactions I feel like

Reply:
Do we want to allow `set_header("header", None)` ?

Reply:
that would be an `Error` I'd say

Reply:
If so, it should not be part of the enum

Reply:
I don't fully agree but I can see the sense in that

Reply:
basic idea with rust enums vs C enums - do not encode "error conditions" in it, like "uninitialized" or "error". Only encode what a user should be allowed to pass in. For everything else, there is an error code as part of the result.